Matrix.SynapseInterop.Worker.FederationSender/TransactionQueue.cs

```diff
@@ -0,0 +1,507 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Transactions;
+using Matrix.SynapseInterop.Common;
+using Matrix.SynapseInterop.Database;
+using Matrix.SynapseInterop.Database.Models;
+using Matrix.SynapseInterop.Replication.Structures;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Extensions.Internal;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Matrix.SynapseInterop.Worker.FederationSender
+{
+    public class TransactionQueue
+    {
+        private const int MAX_PDUS_PER_TRANSACTION = 50;
+        private const int MAX_EDUS_PER_TRANSACTION = 100;
+        private const int MAX_BACKOFF_SECS = 60*60*24;
+        private FederationClient client;
+        private Dictionary<string, PresenceState> userPresence;
+        private Task presenceProcessing;
+        private Task eventsProcessing;
+        private Dictionary<string, Task> destOngoingTrans;
+        private Dictionary<string, Transaction> destPendingTransactions;
+        private SigningKey signingKey;
+        private string serverName;
+        private string connString;
+        private int txnId;
+        private int lastEventPoke;
+
+        private Dictionary<string, long> destLastDeviceMsgStreamId;
+        private Dictionary<string, long> destLastDeviceListStreamId;
+
+        public TransactionQueue(string serverName, string connectionString, SigningKey key)
+        {
+            client = new FederationClient(serverName, key);
```

Comment:
lint that doesn't exist: assign variables where they are defined, not in the constructor